packages/medusa/src/types/routing.ts

```diff
@@ -1,14 +1,19 @@
+import type { Customer, User } from "../models"
 import type { NextFunction, Request, Response } from "express"
 
-import type { Customer, User } from "../models"
 import type { MedusaContainer } from "./global"
 
 export interface MedusaRequest extends Request {
```

Comment:
Suggestion: shouldn't we break this request in two, one for the authenticated route and one for the non authenticated route?

I am thinking of something like the following:
```ts
export interface MedusaRequest<Body = unknown> extends Request {
  validatedBody: Body
  validatedQuery: RequestQueryFields & Record<string, unknown>
  listConfig: FindConfig<unknown>
  retrieveConfig: FindConfig<unknown>
  filterableFields: Record<string, unknown>
  allowedProperties: string[]
  includes?: Record<string, boolean>
  errors: string[]
}

export interface AuthenticatedMedusaRequest<Body = never>
  extends MedusaRequest<Body> {
  user: (User | Customer) & { customer_id?: string; userId?: string }
  auth: {
    actor_id: string
    auth_user_id: string
    app_metadata: Record<string, any>
    scope: string
  }
}
```

and later on we can probably rm the express augmentation too

cc @olivermrbl

Reply:
When we have 2.0 done, I'd remove `user` from here and just base everything on what's in `auth`.

Maybe worth keeping in mind that we should never user `req.user` in api-v2